src/flow/ext_multiroute_flow.jl

```diff
@@ -1,256 +1,251 @@
+    a2::R           # slope passing by point 2
+    ) where T<:AbstractFloat where R<:Real
+
+    (a1 == a2) && return -1., -1. # result will be ignored in other intersection method
+    b1 = y1 - a1 * x1
+    b2 = y2 - a2 * x2
+    x = (b2 - b1) / (a1 - a2)
+    y = a1 * x + b1
+    return x, y
+end
 
-# Compute the (expected) intersection of two lines
-function intersection{T<:AbstractFloat, R<:Real}(
-  x1::T,          # x coordinate of point 1
-  y1::T,          # y coordinate of point 1
-  a1::Int,        # slope passing by point 1
-  x2::T,          # x coordinate of point 2
-  y2::T,          # y coordinate of point 2
-  a2::R           # slope passing by point 2
-  )
 
-  (a1 == a2) && return -1., -1. # result will be ignored in other intersection method
-  b1 = y1 - a1 * x1
-  b2 = y2 - a2 * x2
-  x = (b2 - b1) / (a1 - a2)
-  y = a1 * x + b1
-  return x, y
-end
-# Compute the intersection between a set of segment and a line of slope k passing by the origin
-function intersection{T<:AbstractFloat, R<:Real}(
-  points::Vector{Tuple{T, T, Int}},  # vector of breaking points
-  k::R                               # number of routes (slope of the line)
-  )
-  λ = points[1][1] # Connectivity
+"""
+    intersection(points, k)
 
-  # Loop over the segments (pair of breaking points)
-  for (id, p) in enumerate(points[1:(end - 1)])
-    if id == 1
-      (k ≈ λ) && return points[2]
-    else
-      x, y = intersection(p[1], p[2], p[3], 0., 0., k)
-      (p[1] ≤ x ≤ points[id + 1][1]) && return x, y
+Return the intersection of a set of line segments and a line of slope `k`
+passing by the origin. Segments are defined as a triple (x, y, slope).
+"""
+function intersection(
+    points::Vector{Tuple{T, T, I}},  # vector of breaking points
+    k::R                               # number of routes (slope of the line)
+    ) where T<:AbstractFloat where I<:Integer where R<:Real
+    λ = points[1][1] # Connectivity
+
+    # Loop over the segments (pair of breaking points)
+    for (id, p) in enumerate(points[1:(end - 1)])
+        if id == 1
+            (k ≈ λ) && return points[2]
+        else
+            x, y = intersection(p[1], p[2], p[3], 0., 0., k)
+            (p[1] ≤ x ≤ points[id + 1][1]) && return x, y
+        end
     end
-  end
-  p = points[end]
-  return intersection(p[1], p[2], p[3], 0., 0., k)
+    p = points[end]
+    return intersection(p[1], p[2], p[3], 0., 0., k)
 end
 
 """
-Redefinition of ≈ (isapprox) for a pair of points
-Requires argument:
-  a::Tuple{T, T},         # Point A with floating coordinates
-  b::Tuple{T, T}          # Point B with floating coordinates
+    ≈(a, b)
+
+Redefinition of ≈ (isapprox) for a pair of points `a` and `b`.
 """
-function ≈{T<:AbstractFloat}(
-  a::Tuple{T, T},         # Point A with floating coordinates
-  b::Tuple{T, T}          # Point B with floating coordinates
-  )
-  return a[1] ≈ b[1] && a[2] ≈ b[2]
-end
+≈(a::Tuple{T, T}, b::Tuple{T, T}) where T <: AbstractFloat =
```

Comment:
sure, people have tried adding arithmetic to tuples in a bunch of places, but it has surprising side effects because the language uses tuples for argument lists and return values